io/zenoh-transport/src/unicast/link.rs

```diff
@@ -38,14 +38,22 @@ pub(crate) struct TransportLinkUnicastConfig {
     pub(crate) is_compression: bool,
 }
 
-#[derive(Clone, PartialEq, Eq)]
+#[derive(PartialEq, Eq)]
 pub(crate) struct TransportLinkUnicast {
-    pub(crate) link: LinkUnicast,
+    pub(crate) link: Arc<LinkUnicast>,
```

Comment:
Why un `Arc`? [`LinkUnicast`](https://github.com/eclipse-zenoh/zenoh/blob/eb60dd09b6c67ea1d1d0044ff6531d2e6821d4bf/io/zenoh-link-commons/src/unicast.rs#L39) already contains an `Arc`. This introduce an additional indirection that I don't think is needed...

Reply:
`LinkUnicast` is not `Clone` anymore (it has never been `Copy`). Still it holds an `Arc` internally and here we are introducing an additional indirection that is traversed every time `send` is called. That means, every time we send a message, we pay the cost of the additional indirection... In any case, we are in the situation of having an `Arc<Arc<..>>`...

Reply:
Oh, I mistyped

Reply:
The only way to solve this issue is to get back to `Clone` in `LinkUnicast`, I think I don't have another options. Yes, you are right that we're gonna pay the cost on `send` that is not good